### PR TITLE
[MINI-6077] interface primary seconday color

### PR DESCRIPTION
**Pull request by munir-rakuten**
# Description
Changes

- Added `MiniAppMessageBridge.getHostAppThemeColors` in MiniAppMessageBridge for retrieving primary and secondary theme color.
- Added qa settings to test primary and secondary color.

## Links
MINI-6077

# Checklist
- [x] I have read the [contributing guidelines](https://github.com/rakutentech/android-miniapp/blob/master/.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](https://github.com/rakutentech/android-miniapp/blob/master/.github/CONTRIBUTING.md#sdk-development-learning-path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/hostenvironment/HostThemeColor.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+import androidx.annotation.Keep
+
+/**
+ * Represents the color theme of the host app.
+ * @property primaryColor The primary color of the host app.
+ * @property secondaryColor The secondary color of the host app.
+ */
+@Keep
+data class HostThemeColor(
```
Comment (GondiTilak) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/hostenvironment/HostThemeColor.kt:11]: Can we have it same as in iOS - "HostAppThemeColors" for this class name?

File: miniapp/src/test/java/com/rakuten/tech/mobile/miniapp/TestData.kt [hunk 2]
```diff
@@ -40,6 +41,7 @@ internal const val TEST_HA_SUBSCRIPTION_KEY = "test_subscription_key"
 
 internal const val TEST_CALLBACK_ID = "test_callback_id"
 internal const val TEST_CALLBACK_VALUE = "test_callback_value"
+internal val TEST_CALLBACK_THEME = HostThemeColor("", "")
```
Comment (GondiTilak) [inline miniapp/src/test/java/com/rakuten/tech/mobile/miniapp/TestData.kt:44]: Same here for class name - "HostAppThemeColors"

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/userdata/QASettingsActivity.kt [hunk 3]
```diff
@@ -308,6 +314,17 @@ class QASettingsActivity : BaseActivity() {
             settings.maxStorageSizeLimitInBytes = binding.edtMaxStorageLimit.text.toString()
         }
 
+        //Save color theme
+        var primaryColor = ""
+        var secondaryColor = ""
+        if (!binding.edtPrimaryColor.text.isNullOrEmpty()) {
+            primaryColor = binding.edtPrimaryColor.text.toString()
+        }
+        if (!binding.edtSecondaryColor.text.isNullOrEmpty()) {
```
Comment (GondiTilak) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/userdata/QASettingsActivity.kt:323]: I think we need not add the null or empty check for secondaryColor, we can send the default #FFFFFF - white color, if this makes sense.

Comment (munir-rakuten) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/userdata/QASettingsActivity.kt:323]: i think we don't need to send a default color, just send a empty string if no value is set, sample app will handle that, It's just a check in the demo app if nothing is in settings just simply assign a empty string. thanks for you suggestion though.

**Review by rleojoseph — APPROVED** [review]
Looks good to me :+1:

**Review by GondiTilak — APPROVED** [review]

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/hostenvironment/HostThemeColor.kt (hunk 1 above)
Comment (GondiTilak) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/hostenvironment/HostThemeColor.kt]: Thank you for the changes. 👍🏽

**Pull request merged**